### ISSUE TITLE
var self = this; this.x becomes this after compiling

### ISSUE THREAD

**Issue by cvissie** [google/closure-compiler#1742]
This is not a bug and shouldn't be flagged as such.

I'm merely wondering in the following snippet of code:

```javascript
    var self = this;
    var buffer = self.buffer;

    self.currentFrame++;
    self.currentTime = self.currentFrame / self.pictureRate;

    buffer.advance(10); // skip temporalReference
    self.pictureCodingType = buffer.getBits(3);
    buffer.advance(16); // skip vbv_delay

    var pictureCodingType = self.pictureCodingType;

    // Skip B and D frames or unknown coding type
    if (pictureCodingType <= 0 || pictureCodingType >= PICTURE_TYPE_B) {
      return;
    }

    // full_pel_forward, forward_f_code
    if (pictureCodingType === PICTURE_TYPE_P) {
      self.fullPelForward = buffer.getBits(1);
      self.forwardFCode = buffer.getBits(3);
      if (self.forwardFCode === 0) {
        // Ignore picture with zero forward_f_code
        return;
      }
      self.forwardRSize = self.forwardFCode - 1;
      self.forwardF = 1 << self.forwardRSize;
    }

    var code = 0;
    do {
      code = buffer.findNextMPEGStartCode();
    } while (code === START_EXTENSION || code === START_USER_DATA);


    while (code >= START_SLICE_FIRST && code <= START_SLICE_LAST) {
      self.decodeSlice((code & 0x000000FF));
      code = buffer.findNextMPEGStartCode();
    }

    // We found the next start code; rewind 32bits and let the main loop handle it.
    buffer.rewind(32);

    // Record this frame, if the recorder wants it
    self.recordFrameFromCurrentBuffer();

    if (skipOutput !== DECODE_SKIP_OUTPUT) {
      self.renderFrame();

      if (self.externalDecodeCallback) {
        self.externalDecodeCallback(this, self.canvas);
      }
    }

    // If this is a reference picutre then rotate the prediction pointers
    if (pictureCodingType === PICTURE_TYPE_I || pictureCodingType === PICTURE_TYPE_P) {
      var tmpY = self.forwardY;
      var tmpY32 = self.forwardY32;
      var tmpCr = self.forwardCr;
      var tmpCr32 = self.forwardCr32;
      var tmpCb = self.forwardCb;
      var tmpCb32 = self.forwardCb32;

      self.forwardY = self.currentY;
      self.forwardY32 = self.currentY32;
      self.forwardCr = self.currentCr;
      self.forwardCr32 = self.currentCr32;
      self.forwardCb = self.currentCb;
      self.forwardCb32 = self.currentCb32;

      self.currentY = tmpY;
      self.currentY32 = tmpY32;
      self.currentCr = tmpCr;
      self.currentCr32 = tmpCr32;
      self.currentCb = tmpCb;
      self.currentCb32 = tmpCb32;
    }
```

After compiled using simple optimizations becomes the following:

```javascript
var c=this.buffer;this.currentFrame++;this.currentTime=this.currentFrame/this.pictureRate;c.advance(10);this.pictureCodingType=c.getBits(3);c.advance(16);var b=this.pictureCodingType;if(!(0>=b||b>=N)){if(b===C){this.fullPelForward=
c.getBits(1);this.forwardFCode=c.getBits(3);if(0===this.forwardFCode)return;this.forwardRSize=this.forwardFCode-1;this.forwardF=1<<this.forwardRSize}var e=0;do e=c.findNextMPEGStartCode();while(e===O||e===P);for(;e>=Q&&e<=R;)this.decodeSlice(e&255),e=c.findNextMPEGStartCode();c.rewind(32);this.recordFrameFromCurrentBuffer();a!==D&&(this.renderFrame(),this.externalDecodeCallback&&this.externalDecodeCallback(this,this.canvas));if(b===B||b===C){a=this.forwardY;var c=this.forwardY32,b=this.forwardCr,
e=this.forwardCr32,m=this.forwardCb,d=this.forwardCb32;this.forwardY=this.currentY;this.forwardY32=this.currentY32;this.forwardCr=this.currentCr;this.forwardCr32=this.currentCr32;this.forwardCb=this.currentCb;this.forwardCb32=this.currentCb32;this.currentY=a;this.currentY32=c;this.currentCr=b;this.currentCr32=e;this.currentCb=m;this.currentCb32=d}}
```

Nothing wrong. I'm just wondering why does closure ignore the var self = this; statement and replace it with this.xxx everywhere.

Is it faster for some reason? It definitely makes the code bigger.

**Comment by Dominator008** [comment]
My guess is that the current output is actually smaller after gzipping.
